tests/testthat/test-infrastructure.r

```diff
@@ -59,3 +59,68 @@ test_that("use_data", {
     expect_equal(local_test_data_item_to_save, 42L)
   })
 })
+
+test_that("use_package", {
```

Comment:
I'd do one test for proper functionality, and one test for idempotency.

The problem with the current test is that it is so long that I can't easily understand the goal in a glance. That's a problem because I have to maintain this code, and if `use_package()` needs to change in the future, I need to understand what the tests actually do so that I don't break existing behaviour.